components/brave_today/browser/brave_today_service.cc

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_today/browser/brave_today_service.h"
+
+
+#include <memory>
+#include <utility>
+
+#include "base/metrics/histogram_macros.h"
+#include "brave/components/brave_ads/browser/ads_service.h"
+#include "brave/components/brave_today/common/pref_names.h"
+#include "brave/components/p3a/brave_p3a_service.h"
+#include "brave/components/weekly_storage/weekly_storage.h"
+#include "components/pref_registry/pref_registry_syncable.h"
+
+namespace brave_today {
+
+// static
+void BraveTodayService::RegisterProfilePrefs(
```

Comment:
mb this is BraveTodayP3AService or something, since it is mostly devoted to P3A?